src/main/java/org/llorllale/cactoos/matchers/IsMatcherOf.java

```diff
@@ -0,0 +1,180 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Matcher to test {@link org.hamcrest.Matcher} objects.
+ *
+ * <p>Here is a few examples how we are testing matchers
+ *  without/with {@link IsMatcherOf}:</p>
+ * <pre>{@code
+ *  @Test
+ *  public void matchesSafelyWithout() {
+ *      final Description expected = new StringDescription();
+ *      final Description actual = new StringDescription();
+ *      final HasValues<Integer> hvalues = new HasValues<>(5);
+ *      hvalues.describeTo(expected);
+ *      MatcherAssert.assertThat(
+ *          "The testing status is 'false'",
+ *          hvalues.matchesSafely(new ListOf<>(1, 2, 3), actual),
+ *          new IsEqual<>(false)
+ *      );
+ *      MatcherAssert.assertThat(
+ *          "The actual result message in hamcrest terms is correct",
+ *          actual.toString(),
+ *          new IsEqual<>("The function applied to <[1, 2, 3]> is failed")
+ *      );
+ *  }
+ *
+ *  @Test
+ *  public void matchesSafelyWith() {
+ *      MatcherAssert.assertThat(
+ *          new HasValues<>(5),
+ *          new IsMatcherOf<>(
+ *              false, new ListOf<>(1, 2, 3),
+ *               "The function applied to <[1, 2, 3]> is failed"
+ *          )
+ *       );
+ *   }
+ *
+ * }</pre>
+ *
+ * @param <X> Type of item.
+ * @since 1.0.0
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class IsMatcherOf<X> extends
+    TypeSafeDiagnosingMatcher<Matcher<X>> {
+
+    /**
+     * The expected testing status within matcher.
+     */
+    private final boolean status;
+
+    /**
+     * The expected message with testing details from target matcher.
+     */
+    private final String exp;
+
+    /**
+     * The actual message with testing details from target matcher.
+     */
+    private final String act;
+
+    /**
+     * The testing arguments for the target matcher.
+     */
+    private final X margs;
+
+    /**
+     * Ctor.
+     * @param status The expected testing status within matcher.
+     * @param margs The testing arguments for the matcher.
+     */
+    public IsMatcherOf(final boolean status, final X margs) {
```

Comment:
@dgroup remove this ctor. We should encourage our users to provide assertion messages